htdocs/editor_tab.js

```diff
@@ -1477,6 +1494,19 @@ var editor = function () {
                     return promise;
                 });
         },
+        highlight_imported_notebooks: function(notebooks) {
+
+            var nodes = _.map(_.isArray(notebooks) ? notebooks : [notebooks], function(notebook) {
+                return $tree_.tree('getNodeById', node_id('interests', username_, notebook.id));
+            });
+
+            // get promises:
+            nodes.map(function(node) {
+                return highlight_imported(node);
+            }).reduce(function(cur, next) {
+                return cur.then(next);
+            }, Promise.resolve()).then(function() {});
```

Comment:
That is an extremely clever map-reduce promise! I didn't know you could do that, but I guess I shouldn't be surprised, since functional programming always has a way. This is what [`promise_sequence`](https://github.com/att/rcloud/blob/develop/htdocs/js/utils.js#L11-L20) does.